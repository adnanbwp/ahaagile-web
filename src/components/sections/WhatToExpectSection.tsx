import { Card } from '@/components/ui/card'
import { Search, Target, ArrowRight } from '@/lib/icons'

interface BenefitCardProps {
  icon: React.ComponentType<{ className?: string }>
  title: string
  description: string
}

const BenefitCard = ({ icon: Icon, title, description }: BenefitCardProps) => (
  <Card className="group relative bg-white border-gray-200 shadow-brand hover:shadow-accent transition-all duration-300 hover:scale-105 transform-gpu">
    <div className="p-8 text-center">
      <div className="flex flex-col items-center space-y-6">
        {/* Icon Container with Gradient Background */}
        <div className="relative">
          <div className="w-16 h-16 rounded-full bg-gradient-accent flex items-center justify-center group-hover:scale-110 transition-transform duration-300">
            <Icon className="w-8 h-8 text-white" />
          </div>
          {/* Backdrop blur effect on hover */}
          <div className="absolute inset-0 rounded-full bg-white/10 backdrop-blur-sm opacity-0 group-hover:opacity-100 transition-opacity duration-300"></div>
        </div>
        
        {/* Content */}
        <div className="space-y-3">
          <h3 className="font-heading font-semibold text-xl text-primary-800 group-hover:text-primary-900 transition-colors duration-300">
            {title}
          </h3>
          <p className="text-secondary-600 leading-relaxed font-serif text-base">
            {description}
          </p>
        </div>
      </div>
    </div>
  </Card>
)

export default function WhatToExpectSection() {
  return (
    <section className="relative py-16 lg:py-24 bg-gradient-to-br from-secondary-50 via-white to-primary-50 overflow-hidden">
      {/* Background Effects */}
      <div className="absolute top-20 left-10 w-32 h-32 bg-accent-400/5 rounded-full blur-xl"></div>
      <div className="absolute bottom-20 right-10 w-48 h-48 bg-primary-600/5 rounded-full blur-2xl"></div>
      
      <div className="relative container mx-auto px-6 lg:px-8">
        <div className="max-w-6xl mx-auto">
          {/* Section Header */}
          <div className="text-center mb-16">
            <h2 className="font-heading text-3xl lg:text-4xl font-bold text-primary-800 mb-6">
              What to Expect in Your Consultation
            </h2>
            <p className="text-lg text-secondary-600 font-serif max-w-2xl mx-auto leading-relaxed">
              Our structured approach ensures you get maximum value from every consultation session
            </p>
          </div>
          
          {/* Benefit Cards Grid */}
          <div className="grid grid-cols-1 md:grid-cols-3 gap-8 lg:gap-12">
            <BenefitCard
              icon={Search}
              title="Comprehensive Assessment"
              description="Analysis of current workflows and identification of automation opportunities tailored to your specific business needs"
            />
            <BenefitCard
              icon={Target}
              title="Custom Strategy"
              description="Development of tailored automation roadmap based on your specific requirements and organizational goals"
            />
            <BenefitCard
              icon={ArrowRight}
              title="Clear Next Steps"
              description="Actionable implementation plan and ongoing support guidance to ensure successful automation deployment"
            />
          </div>
        </div>
      </div>
    </section>
  )
}