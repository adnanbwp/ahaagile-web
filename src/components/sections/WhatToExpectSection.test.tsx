import { render, screen } from '@testing-library/react'
import WhatToExpectSection from './WhatToExpectSection'

// Mock Lucide icons
jest.mock('lucide-react', () => ({
  Search: ({ className }: { className?: string }) => (
    <div data-testid="search-icon" className={className}>Search Icon</div>
  ),
  Target: ({ className }: { className?: string }) => (
    <div data-testid="target-icon" className={className}>Target Icon</div>
  ),
  ArrowRight: ({ className }: { className?: string }) => (
    <div data-testid="arrow-right-icon" className={className}>Arrow Right Icon</div>
  ),
}))

describe('WhatToExpectSection', () => {
  beforeEach(() => {
    render(<WhatToExpectSection />)
  })

  describe('Section Structure', () => {
    it('renders the main section with correct heading', () => {
      const heading = screen.getByRole('heading', { 
        name: /what to expect in your consultation/i 
      })
      expect(heading).toBeInTheDocument()
      expect(heading).toHaveClass('font-heading', 'text-3xl', 'lg:text-4xl', 'font-bold', 'text-primary-800')
    })

    it('renders the section description', () => {
      const description = screen.getByText(/our structured approach ensures you get maximum value/i)
      expect(description).toBeInTheDocument()
      expect(description).toHaveClass('text-lg', 'text-secondary-600', 'font-serif')
    })

    it('applies correct background gradient and styling', () => {
      const section = screen.getByText(/what to expect in your consultation/i).closest('section')
      expect(section).toHaveClass(
        'relative', 
        'py-16', 
        'lg:py-24', 
        'bg-gradient-to-br', 
        'from-secondary-50', 
        'via-white', 
        'to-primary-50'
      )
    })
  })

  describe('Benefit Cards', () => {
    it('renders all three benefit cards', () => {
      // Assessment card
      expect(screen.getByText('Comprehensive Assessment')).toBeInTheDocument()
      expect(screen.getByText(/analysis of current workflows and identification/i)).toBeInTheDocument()
      expect(screen.getByTestId('search-icon')).toBeInTheDocument()

      // Strategy card
      expect(screen.getByText('Custom Strategy')).toBeInTheDocument()
      expect(screen.getByText(/development of tailored automation roadmap/i)).toBeInTheDocument()
      expect(screen.getByTestId('target-icon')).toBeInTheDocument()

      // Next Steps card
      expect(screen.getByText('Clear Next Steps')).toBeInTheDocument()
      expect(screen.getByText(/actionable implementation plan and ongoing support/i)).toBeInTheDocument()
      expect(screen.getByTestId('arrow-right-icon')).toBeInTheDocument()
    })

    it('applies correct styling to card titles', () => {
      const titles = [
        'Comprehensive Assessment',
        'Custom Strategy', 
        'Clear Next Steps'
      ]

      titles.forEach(title => {
        const titleElement = screen.getByText(title)
        expect(titleElement).toHaveClass(
          'font-heading',
          'font-semibold', 
          'text-xl',
          'text-primary-800'
        )
      })
    })

    it('applies correct styling to card descriptions', () => {
      const descriptions = [
        /analysis of current workflows and identification/i,
        /development of tailored automation roadmap/i,
        /actionable implementation plan and ongoing support/i
      ]

      descriptions.forEach(description => {
        const descElement = screen.getByText(description)
        expect(descElement).toHaveClass(
          'text-secondary-600',
          'leading-relaxed',
          'font-serif',
          'text-base'
        )
      })
    })

    it('renders card icons with correct styling', () => {
      const icons = ['search-icon', 'target-icon', 'arrow-right-icon']
      
      icons.forEach(iconTestId => {
        const icon = screen.getByTestId(iconTestId)
        expect(icon).toHaveClass('w-8', 'h-8', 'text-white')
      })
    })
  })

  describe('Interactive Effects', () => {
    it('applies hover and transition classes to cards', () => {
      // Get the cards by finding elements with Card component classes
      const cardTitles = ['Comprehensive Assessment', 'Custom Strategy', 'Clear Next Steps']
      
      cardTitles.forEach(title => {
        const card = screen.getByText(title).closest('.group')
        expect(card).toHaveClass(
          'group',
          'hover:shadow-accent',
          'hover:scale-105',
          'transition-all',
          'duration-300'
        )
      })
    })

    it('applies hover effects to icon containers', () => {
      // Icon containers should have gradient backgrounds and hover scale effects
      const section = screen.getByText(/what to expect in your consultation/i).closest('section')
      expect(section).toBeInTheDocument()
      
      // Test that hover classes are present in the component
      const componentHTML = section!.innerHTML
      expect(componentHTML).toContain('group-hover:scale-110')
      expect(componentHTML).toContain('bg-gradient-accent')
      expect(componentHTML).toContain('backdrop-blur-sm')
    })
  })

  describe('Responsive Design', () => {
    it('applies responsive grid layout', () => {
      const section = screen.getByText(/what to expect in your consultation/i).closest('section')
      const gridContainer = section!.querySelector('.grid')
      expect(gridContainer).toHaveClass(
        'grid',
        'grid-cols-1',
        'md:grid-cols-3',
        'gap-8',
        'lg:gap-12'
      )
    })

    it('applies responsive typography classes', () => {
      const heading = screen.getByRole('heading', { 
        name: /what to expect in your consultation/i 
      })
      expect(heading).toHaveClass('text-3xl', 'lg:text-4xl')
    })

    it('applies responsive padding classes', () => {
      const section = screen.getByText(/what to expect in your consultation/i).closest('section')
      expect(section).toHaveClass('py-16', 'lg:py-24')
    })
  })

  describe('Accessibility', () => {
    it('uses proper semantic HTML structure', () => {
      // Main heading should be h2
      const mainHeading = screen.getByRole('heading', { 
        name: /what to expect in your consultation/i 
      })
      expect(mainHeading.tagName).toBe('H2')

      // Card titles should be h3
      const cardTitles = screen.getAllByRole('heading', { level: 3 })
      expect(cardTitles).toHaveLength(3)
    })

    it('provides meaningful text content for screen readers', () => {
      // All cards should have descriptive content
      expect(screen.getByText(/analysis of current workflows and identification of automation opportunities/i)).toBeInTheDocument()
      expect(screen.getByText(/development of tailored automation roadmap based on your specific requirements/i)).toBeInTheDocument()
      expect(screen.getByText(/actionable implementation plan and ongoing support guidance/i)).toBeInTheDocument()
    })

    it('maintains proper heading hierarchy', () => {
      const headings = screen.getAllByRole('heading')
      
      // Should have one h2 (main heading) and three h3s (card titles)
      const h2Elements = headings.filter(h => h.tagName === 'H2')
      const h3Elements = headings.filter(h => h.tagName === 'H3')
      
      expect(h2Elements).toHaveLength(1)
      expect(h3Elements).toHaveLength(3)
    })
  })

  describe('Design System Integration', () => {
    it('uses correct brand colors from design system', () => {
      const heading = screen.getByRole('heading', { 
        name: /what to expect in your consultation/i 
      })
      expect(heading).toHaveClass('text-primary-800')

      const description = screen.getByText(/our structured approach ensures/i)
      expect(description).toHaveClass('text-secondary-600')
    })

    it('uses correct font families from design system', () => {
      const heading = screen.getByRole('heading', { 
        name: /what to expect in your consultation/i 
      })
      expect(heading).toHaveClass('font-heading')

      const description = screen.getByText(/our structured approach ensures/i)
      expect(description).toHaveClass('font-serif')
    })

    it('integrates with shadcn/ui Card component', () => {
      // Cards should be rendered (testing presence of card content)
      expect(screen.getByText('Comprehensive Assessment')).toBeInTheDocument()
      expect(screen.getByText('Custom Strategy')).toBeInTheDocument()
      expect(screen.getByText('Clear Next Steps')).toBeInTheDocument()
    })
  })
})

describe('WhatToExpectSection Snapshot', () => {
  it('matches snapshot to prevent visual regression', () => {
    const { container } = render(<WhatToExpectSection />)
    expect(container.firstChild).toMatchSnapshot()
  })
})