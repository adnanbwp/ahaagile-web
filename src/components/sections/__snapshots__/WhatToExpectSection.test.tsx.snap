// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhatToExpectSection Snapshot matches snapshot to prevent visual regression 1`] = `
<section
  class="relative py-16 lg:py-24 bg-gradient-to-br from-secondary-50 via-white to-primary-50 overflow-hidden"
>
  <div
    class="absolute top-20 left-10 w-32 h-32 bg-accent-400/5 rounded-full blur-xl"
  />
  <div
    class="absolute bottom-20 right-10 w-48 h-48 bg-primary-600/5 rounded-full blur-2xl"
  />
  <div
    class="relative container mx-auto px-6 lg:px-8"
  >
    <div
      class="max-w-6xl mx-auto"
    >
      <div
        class="text-center mb-16"
      >
        <h2
          class="font-heading text-3xl lg:text-4xl font-bold text-primary-800 mb-6"
        >
          What to Expect in Your Consultation
        </h2>
        <p
          class="text-lg text-secondary-600 font-serif max-w-2xl mx-auto leading-relaxed"
        >
          Our structured approach ensures you get maximum value from every consultation session
        </p>
      </div>
      <div
        class="grid grid-cols-1 md:grid-cols-3 gap-8 lg:gap-12"
      >
        <div
          class="rounded-lg border text-card-foreground shadow-sm group relative bg-white border-gray-200 shadow-brand hover:shadow-accent transition-all duration-300 hover:scale-105 transform-gpu"
        >
          <div
            class="p-8 text-center"
          >
            <div
              class="flex flex-col items-center space-y-6"
            >
              <div
                class="relative"
              >
                <div
                  class="w-16 h-16 rounded-full bg-gradient-accent flex items-center justify-center group-hover:scale-110 transition-transform duration-300"
                >
                  <div
                    class="w-8 h-8 text-white"
                    data-testid="search-icon"
                  >
                    Search Icon
                  </div>
                </div>
                <div
                  class="absolute inset-0 rounded-full bg-white/10 backdrop-blur-sm opacity-0 group-hover:opacity-100 transition-opacity duration-300"
                />
              </div>
              <div
                class="space-y-3"
              >
                <h3
                  class="font-heading font-semibold text-xl text-primary-800 group-hover:text-primary-900 transition-colors duration-300"
                >
                  Comprehensive Assessment
                </h3>
                <p
                  class="text-secondary-600 leading-relaxed font-serif text-base"
                >
                  Analysis of current workflows and identification of automation opportunities tailored to your specific business needs
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="rounded-lg border text-card-foreground shadow-sm group relative bg-white border-gray-200 shadow-brand hover:shadow-accent transition-all duration-300 hover:scale-105 transform-gpu"
        >
          <div
            class="p-8 text-center"
          >
            <div
              class="flex flex-col items-center space-y-6"
            >
              <div
                class="relative"
              >
                <div
                  class="w-16 h-16 rounded-full bg-gradient-accent flex items-center justify-center group-hover:scale-110 transition-transform duration-300"
                >
                  <div
                    class="w-8 h-8 text-white"
                    data-testid="target-icon"
                  >
                    Target Icon
                  </div>
                </div>
                <div
                  class="absolute inset-0 rounded-full bg-white/10 backdrop-blur-sm opacity-0 group-hover:opacity-100 transition-opacity duration-300"
                />
              </div>
              <div
                class="space-y-3"
              >
                <h3
                  class="font-heading font-semibold text-xl text-primary-800 group-hover:text-primary-900 transition-colors duration-300"
                >
                  Custom Strategy
                </h3>
                <p
                  class="text-secondary-600 leading-relaxed font-serif text-base"
                >
                  Development of tailored automation roadmap based on your specific requirements and organizational goals
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="rounded-lg border text-card-foreground shadow-sm group relative bg-white border-gray-200 shadow-brand hover:shadow-accent transition-all duration-300 hover:scale-105 transform-gpu"
        >
          <div
            class="p-8 text-center"
          >
            <div
              class="flex flex-col items-center space-y-6"
            >
              <div
                class="relative"
              >
                <div
                  class="w-16 h-16 rounded-full bg-gradient-accent flex items-center justify-center group-hover:scale-110 transition-transform duration-300"
                >
                  <div
                    class="w-8 h-8 text-white"
                    data-testid="arrow-right-icon"
                  >
                    Arrow Right Icon
                  </div>
                </div>
                <div
                  class="absolute inset-0 rounded-full bg-white/10 backdrop-blur-sm opacity-0 group-hover:opacity-100 transition-opacity duration-300"
                />
              </div>
              <div
                class="space-y-3"
              >
                <h3
                  class="font-heading font-semibold text-xl text-primary-800 group-hover:text-primary-900 transition-colors duration-300"
                >
                  Clear Next Steps
                </h3>
                <p
                  class="text-secondary-600 leading-relaxed font-serif text-base"
                >
                  Actionable implementation plan and ongoing support guidance to ensure successful automation deployment
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
